src/lib/utils.rs

```diff
@@ -62,6 +62,18 @@ pub fn segment_kind_to_fastq_id(kind: &SegmentType) -> char {
     }
 }
 
+/// Determine the output file name for the given sample, segment type, and type number.
+pub fn filename(sample: &SampleMetadata, kind: &SegmentType, type_number: i32) -> String {
```

Comment:
i32 seems like an odd choice - I would have expected a `u` something, e.g. `u16` or `u32`.  Can we have negative read numbers?